src/Avalonia.Controls.TreeDataGrid/Selection/TreeDataGridRowSelectionModel.cs

```diff
@@ -110,6 +113,99 @@ void ITreeDataGridSelectionInteraction.OnKeyDown(TreeDataGrid sender, KeyEventAr
 
         }
 
+        protected void HandleCharKey(string? text, TreeDataGrid treeDataGrid, int selectedRowIndex)
+        {
+            if (text != null && treeDataGrid.Columns != null)
+            {
+                var typedChar = text[0];
+
+                long now = DateTime.Now.Ticks;
```

Comment:
It's probably very much an edge case but I'm not sure that using `DateTime.Now` is the best idea here because if the system clock is adjusted (e.g. due to a daylight savings time change) then things might go wrong! I think `Stopwatch` or `Environment.TickCount` would be better.